src/Microsoft.VisualStudio.Editor.Razor/DefaultRazorCompletionFactsService.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
```

Comment:
All this logic is refactored from `RazorDirectiveCompletionSource` mostly verbatim